demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -61,7 +61,7 @@ protected void onCreate(Bundle savedInstanceState) {
         );
 
         downloadMigrator = DownloadMigratorBuilder.newInstance(this)
-                .withNotificationConfig(notificationConfig)
+                .withNotification(notificationCreator)
```

Comment:
So now the client has two options.

1. Just use what's requested:
```
downloadMigrator = DownloadMigratorBuilder.newInstance(this)
                .withNotification(new NotificationCreator<MigrationStatus>() {
                    @Override
                    public NotificationInformation createNotification(MigrationStatus notificationPayload) {
                        return null;
                    }

                    @Override
                    public NotificationChannel createNotificationChannel() {
                        return null;
                    }
                })
                .build();
```

2. Use the builder:
```
NotificationCreator<MigrationStatus> notificationCreator = NotificationCreator.Factory.build(
                this,
                "chocolate",
                "Migration notifications",
                notificationCustomizer,
                NotificationManagerCompat.IMPORTANCE_DEFAULT
        );
```

is that ok, expected?

Reply:
Option one is disallowed. The client should only get an instance of `NotificationCreator` by using the factory. We can explain this in Javadoc when we do that task. This is also why `NotificationInformation` is not public.

In fact `NotificationInformation` does not need to be an interface, it can just be a data POJO with getters.

Reply:
let's have a chat on that because it is allowed currently, preventing in through javadocs is not good enough...

Reply:
We have agreed to only expose just one one way of interacting with the notification creator